docs/src/pages/components/data-grid/filtering/filtering.md

```diff
@@ -4,162 +4,240 @@ title: Data Grid - Filtering
 
 # Data Grid - Filtering
 
-<p class="description">Filtering helps to view a subset of the records based on a criteria.</p>
+<p class="description">Easily filter your rows based on one or several criteria.</p>
 
-## Basic filter
+The filters can be modified through grid interface in several ways:
 
-Column filters can be set using the column menu and clicking the Filter menu item.
-Alternatively, if the grid has the toolbar displayed, you just need to click on the Filters button.
+- By opening the column menu and clicking the _Filter_ menu item.
+- By clicking the _Filters_ button in the grid toolbar (if enabled).
 
-The filter applied to a column can be pre-configured using the `filterModel` prop:
+Each column types has its own filter operators.
+The demo below let you explore all the operators for each built-in column type.
+
+_See [the dedicated section](#customize-the-operators) to learn how to create your own custom filter operator._
+
+{{"demo": "pages/components/data-grid/filtering/BasicExampleDataGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Single and multi-filtering
+
+> ⚠️ The `DataGrid` can only filter the rows according to one criterion at the time.
+>
+> To use multi-filtering, you need to upgrade to the [Pro plan](https://mui.com/store/items/material-ui-pro/)
+
+The demo below let you filter the rows according to several criteria at the same time.
+
+{{"demo": "pages/components/data-grid/filtering/BasicExampleDataGridPro.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Pass filters to the grid
+
+### Structure of the model
+
+The full typing details can be found on the [GridFilterModel api page](/api/data-grid/grid-filter-model/).
+
+The filter model is composed of a list of `items` and a `linkOperator`:
+
+#### The `items`
+
+A filter item represents a filtering rule and is composed of several elements:
+
+- `filterItem.columnField`: the field on which we want to apply the rule.
+- `filterItem.value`: the value to look for.
+- `filterItem.operatorValue`: name of the operator method to use (e.g. _contains_), matches the `value` key of the .
+- `filterItem.id` ([<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)): only useful when multiple filters are used.
+
+#### The `linkOperator` [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+The `linkOperator` tells the grid if a row should pass all the items or at least one in order to be considered valid.
+
+In the example below, the rows with `rating > 4` or `isAdmin = true` will be displayed.
+
```

Comment:
This could avoid having to "big" code example with a small line between

```ts
// Example 1: get rows with rating > 4 OR isAdmin = true
const filterModel: GridFilterModel = {
  items: [
    { id: 1, columnField: 'rating', operatorValue: '>', value: '4' },
    { id: 2, columnField: 'isAdmin', operatorValue: 'is', value: 'true' },
  ],
  linkOperator: GridLinkOperator.Or,
};

// Example 2: get rows with rating > 4 AND isAdmin = true
const filterModel: GridFilterModel = {
  items: [
    { id: 1, columnField: 'rating', operatorValue: '>', value: '4' },
    { id: 2, columnField: 'isAdmin', operatorValue: 'is', value: 'true' },
  ],
  linkOperator: GridLinkOperator.And,
};
```

Reply:
I'm wondering if it should be precise that `value` is optional for filters such as "is empty"

Reply:
Done

> I'm wondering if it should be precise that value is optional for filters such as "is empty"

I added a note

```
**Note**: Some operators do not need any value (for instance the `isEmpty` operator of the `string` column).
```

After the filter page split, if we feel like it could be useful, we could add a table with all the details about the built-in operators.
Right now, we rely on the fact that their name is mostly self-explanatory and that their code is simple to understand. But that makes the doc not 100% sufficient.